proxy/http2/Http2ConnectionState.cc

```diff
@@ -1054,7 +1055,7 @@ Http2ConnectionState::send_a_data_frame(Http2Stream *stream, size_t &payload_len
     return HTTP2_SEND_A_DATA_FRAME_NO_PAYLOAD;
   }
 
-  if (stream->is_body_done() && payload_length < available_size) {
+  if (stream->is_body_done() && payload_length <= available_size) {
```

Comment:
IMO, to stop sending 0 payload DATA frame with END_STREAM flag, we need to check read available size of current_reader. Something like this.
https://paste.apache.org/8eVP